nathanjophillips/nathan-phillips-ux-aop
# Unit 3.1 npm scripts local server

## The Purpose
- Gain a basic understanding npm scripts and setting up a local server for hosting
- To apply the use of a package manager to host static html files
- To make my project more usable for testing in the browser for myself and others reviewing the project

## What It Does
1. This feature allows anyone who has access to this project to host it locally on their device in their desired browser for more accurate testing

2. This is done by having added a `package.json` file as a package manager to run pre-written scripts and dependencies

3. The dependencies uses in this feature are `http-server`, `npm`, and `yarn`

4. In the `package.json` file the `script` object runs a `start` command that triggers `http-server` to run my project starting at `index.html` in the `src` folder


### This accomplishes the following requirements given for  [Unit 3.1](https://github.com/phillnab/nathan-phillips-ux-iop/blob/master/issues/unit-3/3.1-npm-scripts.md) :

#### When Accomplished:
- [x] `package.json` file includes `scripts` object with `start` property that runs a server 
- [x] Description of `start` script exists in `package.json`
- [x] `yarn start` runs the server
- [x] `http://localhost:1820` loads an HTML page in a browser

### The changes to look for are:
1. Check for errors in the terminal
2. Check for syntax mistakes in the `package.json` file 
3. Once running check if each page is showing with css applied 
#### Pages being:
- User Profile
- User List 
- Create User 
- Edit user

# How To Test
- Clone Repo [nathan-phillips-ux-iop](https://github.com/phillnab/nathan-phillips-ux-iop)
- Checkout branch `nathan-phillips-ux-iop/unit-3.1-npm-scripts-local-server`
- Make sure `npm` and `yarn` are both installed as dependencies
- Enter the terminal to run command `yarn start`
- Click the <b>top</b> link ` http://127.0.0.1:1820` to enter the hosted pages
- Check to see if page look like the following screenshots

![Screen Shot 2019-03-21 at 1 58 42 PM](https://user-images.githubusercontent.com/29068248/54778060-ba7d2800-4be1-11e9-8760-4d2dcb3a0476.png)
![Screen Shot 2019-03-21 at 1 58 50 PM](https://user-images.githubusercontent.com/29068248/54778065-be10af00-4be1-11e9-9177-2a8aafad7b8c.png)
![Screen Shot 2019-03-21 at 1 58 59 PM](https://user-images.githubusercontent.com/29068248/54778069-c0730900-4be1-11e9-9f85-c2ad0db56948.png)
![Screen Shot 2019-03-21 at 1 59 06 PM](https://user-images.githubusercontent.com/29068248/54778073-c23ccc80-4be1-11e9-985a-6424e696453e.png)

## Checklist

Under penalty of public shaming, I avow that I:

- [x] Referenced applicable issues ()

Browsers tested:
